src/components/common/Textarea.tsx

```diff
@@ -1,37 +1,48 @@
-import React from 'react'
+import React, { useState, forwardRef } from 'react'
+import { cn } from '../../utils/cn'
 
 interface TextareaProps {
   value: string
   onChange: (e: React.ChangeEvent<HTMLTextAreaElement>) => void
   placeholder?: string
-  disabled?: boolean
-  error?: string
   rows?: number
+  className?: string
+  onKeyDown?: (e: React.KeyboardEvent<HTMLTextAreaElement>) => void
 }
 
-export default function Textarea({
-  value,
-  onChange,
-  placeholder = '',
-  disabled = false,
-  error,
-  rows = 4,
-}: TextareaProps) {
-  return (
-    <div className="w-full">
+const Textarea = forwardRef<HTMLTextAreaElement, TextareaProps>(
+  (
+    { value, onChange, placeholder = '', rows = 4, className = '', onKeyDown },
+    ref
+  ) => {
+    const [focused, setFocused] = useState(false)
+
+    const variant = focused ? 'focus' : 'default'
+    const variantMap = {
+      default: 'bg-white text-black border-gray-250',
+      focus: 'border-primary',
```

Comment:
스타일을 주기 위해 포커스라는 상태가 사용되고 있지만, 
focused, blur는 스타일로도 제어가 가능합니다. 
```suggestion
     className={cn(
          'w-full px-4 py-3 text-sm rounded-lg transition-colors resize-none border focus:outline-none',
          'bg-white text-black border-gray-250',
          'focus:border-primary',
          className
        )}
```

Reply:
확실히 불필요한 상태값이 사라지네요